pkg/generators/builder/builder.go

```diff
@@ -161,8 +161,18 @@ func (b *BuilderPatternGenerator) generateSettersForType(sw *generator.SnippetWr
 			if b.isTypeEnabled(m.Type) {
 				sw.Do(setter.GenerateSetterForEmbeddedStruct(m, b.getWrapperType(m.Type)))
 			}
-		case m.Type.Kind == types.Pointer && m.Type.Elem.Kind == types.Builtin:
-			sw.Do(setter.GenerateSetterForPointerToBuiltinType(m))
+		case m.Type.Kind == types.Pointer:
+			pointerType := m.Type.Elem
+			switch pointerType.Kind {
+			case types.Builtin:
+				sw.Do(setter.GenerateSetterForPointerToBuiltinType(m))
+			case types.Struct:
+				if b.isTypeEnabled(pointerType) {
+					sw.Do(setter.GenerateSetterForPrimitiveType(m))
```

Comment:
not sure if calling `GenerateSetterForPrimitiveType` is confusing here. Maybe I should add a function `GenerateSetterForPointerType` that just calls `GenerateSetterForPrimitiveType`

Reply:
I agree that is odd.   I think more simply it can be renamed to `GenerateSetterForType`.  This seems to be the generic setter for a type.